app/scripts/main/controllers/url-selector.controller.js

```diff
@@ -1,163 +1,151 @@
-            var hasCurEtlUrl = _.contains($scope.urlEtlRestBaseList, currentEtlUrl);
-            if (!hasCurEtlUrl) EtlRestServicesSettings.addUrlToList(currentEtlUrl);
-
-            $scope.currentRestBaseUrl = currentUrl;
-            $scope.currentEtlRestBaseUrl = currentEtlUrl;
+(function() {
+  'use strict';
 
-        }
+  /**
+   * @ngdoc function
+   * @name ngAmrsApp.controller:MainCtrl
+   * @description
+   * # MainCtrl
+   * Controller of the ngAmrsApp
+   */
+  angular
+    .module('ngAmrsApp')
+    .controller('UrlSelectorCtrl', UrlSelectorCtrl);
+  UrlSelectorCtrl.$nject = ['$rootScope', '$scope', 'OpenmrsSettings', 'EtlRestServicesSettings',
+    'AuthService', '$state', '$http'
+  ];
 
-        function addUrlToList() {
-            if ($scope.addingUrl) {
-                if ($scope.urlToAdd && $scope.urlToAdd !== '') {
-                    OpenmrsSettings.addUrlToList($scope.urlToAdd);
-                    $scope.currentRestBaseUrl = $scope.urlToAdd;
-                    $scope.addingUrl = false;
-                }
-            }
-            else {
-                $scope.addingUrl = true;
-            }
+  function UrlSelectorCtrl($rootScope, $scope, OpenmrsSettings, EtlRestServicesSettings,
+    AuthService, $state, $http) {
 
-        }
+    $scope.currentMode = '';
 
-        function cancelAddingUrl() {
-            $scope.addingUrl = false;
-            $scope.addingEtlUrl = false;
-        }
+    $scope.currentRestBaseUrl = '';
 
-        function addEtlUrlToList() {
-            if ($scope.addingEtlUrl) {
-                if ($scope.urlEtlToAdd && $scope.urlEtlToAdd !== '') {
-                    EtlRestServicesSettings.addUrlToList($scope.urlEtlToAdd);
-                    $scope.currentEtlRestBaseUrl = $scope.urlEtlToAdd;
-                    $scope.addingEtlUrl = false;
-                }
-            }
-            else {
-                $scope.addingEtlUrl = true;
-            }
-        }
+    $scope.currentEtlRestBaseUrl = '';
 
-        function saveSettings() {
-            EtlRestServicesSettings.setCurrentRestUrlBase($scope.currentEtlRestBaseUrl);
-            OpenmrsSettings.setCurrentRestUrlBase($scope.currentRestBaseUrl);
-            AuthService.logOut();
-        }
+    $scope.urlRestBaseList = [];
+    $scope.urlEtlRestBaseList = [];
+
+    $scope.urlToAdd = '';
+
+    $scope.urlEtlToAdd = '';
+
+    $scope.addingUrl = false;
+
+    $scope.addingEtlUrl = false;
+
+    $scope.preConfiguredSettings = [];
+
+    //methods
+    $scope.addUrlToList = addUrlToList;
+    $scope.addEtlUrlToList = addEtlUrlToList;
+    $scope.saveSettings = saveSettings;
+    $scope.cancelAddingUrl = cancelAddingUrl;
+    $scope.selectConfiguration = selectConfiguration;
+
+    activate();
+
+    function activate() {
+      $http.get('servers.json').
+      success(function(data, status, headers, config) {
+        $scope.preConfiguredSettings = data;
```

Comment:
Move this loading of servers to it's function own function  and put a busy indicator on the ui to indicate that it's loading the templates...